partcad/src/partcad/cache_shape.py

```
@@ -0,0 +1,130 @@
+                key_is_cached_in_files
+                and user_config.cache_memory_double_cache_max_entry_size > 0
+                and data_len > user_config.cache_memory_double_cache_max_entry_size
+            ):
+                # The object is bigger than what we want to store in both caches
+                results[key] = False
+            else:
+                results[key] = True
+
+        return results
+
+    async def read_async(self, hash: CacheHash, keys: list[str]) -> tuple[dict[str, object], dict[str, bool]]:
```

Ruff (0.8.2)

`read_async` is too complex (11 > 10)

(C901)